rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -962,6 +962,76 @@ public void updateCollections(Transaction rskTx) throws IOException {
+            String message = "Proposed federation should be present when processing SVP fund transaction.";
+            logger.warn(message);
+            throw new IllegalStateException(message);
+        }
+        Federation proposedFederation = proposedFederationOpt.get();
+
+        Coin spendableValueFromProposedFederation = bridgeConstants.getSpendableValueFromProposedFederation();
+        BtcTransaction svpFundTransactionUnsigned = createSvpFundTransaction(proposedFederation, spendableValueFromProposedFederation);
+
+        provider.setSvpFundTxHashUnsigned(svpFundTransactionUnsigned.getHash());
+        settleReleaseRequest(svpFundTransactionUnsigned, rskTx.getHash(), spendableValueFromProposedFederation);
+    }
+
+    private BtcTransaction createSvpFundTransaction(Federation proposedFederation, Coin spendableValueFromProposedFederation) throws IOException, InsufficientMoneyException {
+        Wallet activeFederationWallet = getActiveFederationWallet(true);
+
+        BtcTransaction svpFundTransaction = new BtcTransaction(bridgeConstants.getBtcParams());
+        svpFundTransaction.setVersion(BTC_TX_VERSION_2);
+
+        // add outputs to proposed fed and proposed fed with flyover prefix
+        svpFundTransaction.addOutput(spendableValueFromProposedFederation, proposedFederation.getAddress());
+
+        Script proposedFederationWithFlyoverPrefixScriptPubKey = PegUtils.getFederationWithFlyoverPrefixScriptPubKey(bridgeConstants.getProposedFederationFlyoverPrefix(), proposedFederation);
+        svpFundTransaction.addOutput(spendableValueFromProposedFederation, proposedFederationWithFlyoverPrefixScriptPubKey);
+
+        // complete tx with input and change output
+        SendRequest sendRequest = createSvpFundTransactionSendRequest(svpFundTransaction);
+        activeFederationWallet.completeTx(sendRequest);
+
+        return svpFundTransaction;
+    }
+
+    private SendRequest createSvpFundTransactionSendRequest(BtcTransaction transaction) {
+        SendRequest sendRequest = SendRequest.forTx(transaction);
+        sendRequest.changeAddress = getActiveFederationAddress();
+        sendRequest.feePerKb = feePerKbSupport.getFeePerKb();
+        sendRequest.missingSigsMode = Wallet.MissingSigsMode.USE_OP_ZERO;
+        sendRequest.recipientsPayFees = false;
+
+        return sendRequest;
+    }
+
+    private void settleReleaseRequest(BtcTransaction pegoutTransaction, Keccak256 releaseCreationTxHash, Coin requestedAmount) throws IOException {
+        addPegoutToPegoutsWaitingForConfirmations(pegoutTransaction, releaseCreationTxHash);
+        savePegoutTxSigHash(pegoutTransaction);
+        logReleaseRequested(releaseCreationTxHash, pegoutTransaction, requestedAmount);
+        logPegoutTransactionCreated(pegoutTransaction);
+    }
+
+    private void addPegoutToPegoutsWaitingForConfirmations(BtcTransaction pegoutTransaction, Keccak256 releaseCreationTxHash) throws IOException {
```

Comment:
There is another method in this same class with a very similar name and very similar behaviour. Shouldn't we try to consolidate this?

Reply:
theres a new task created to do it and to reuse new settle release request method whenever creating a pegout